### PR TITLE
Added OpenVINO util library

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - Introduced OpenVINO util static library for some private common API

### CODE REVIEW

**Comment by ilya-lavrenov** [comment]
Can we also merge with:
- https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/plugin_api/debug.h
- https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/plugin_api/file_utils.h
- https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/plugin_api/ie_algorithm.hpp
?

**Comment by ilyachur** [comment]
> 
> 
> Can we also merge with:
> 
>     * https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/plugin_api/debug.h
> 
>     * https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/plugin_api/file_utils.h
> 
>     * https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/plugin_api/ie_algorithm.hpp
>       ?

Yes I will merge

File: openvino/util/include/openvino/util/file_util.hpp [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+#endif
+    }
+    static std::string library_prefix() {
+#ifdef _WIN32
+        return {""};
+#else
+        return {"lib"};
+#endif
+    }
+};
```
Comment (ilya-lavrenov) [inline openvino/util/include/openvino/util/file_util.hpp:52]: this was taken from IE, can we remove IE stuff and re-use this one there?
E.i. we remove IE's implementation and use this one. 
For header's only code we can use `using ov::utils::FileUtils`

Comment (ilyachur) [reply openvino/util/include/openvino/util/file_util.hpp:52]: Removed

**Comment by ilyachur** [comment]
@ilya-lavrenov @apankratovantonp  Please take a look

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**